Fix #155018 (convert distributed rst to markdown)

j-silv opened the pull request:
Used [rst2myst tool](https://rst-to-myst.readthedocs.io/en/latest/)

Fixes #155018

Docs comparison (check out the 'new' whenever docs build)

1. distributed.checkpoint ([old](https://docs.pytorch.org/docs/main/distributed.checkpoint.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155528/distributed.checkpoint.html)) 
2. distributed.elastic ([old](https://docs.pytorch.org/docs/main/distributed.elastic.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155528/distributed.elastic.html))
3. distributed.fsdp.fully_shard ([old](https://docs.pytorch.org/docs/main/distributed.fsdp.fully_shard.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155528/distributed.fsdp.fully_shard.html))
4. distributed.optim ([old](https://docs.pytorch.org/docs/main/distributed.optim.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155528/distributed.optim.html))
5. distributed.pipelining ([old](https://docs.pytorch.org/docs/main/distributed.pipelining.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155528/distributed.pipelining.html))

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @svekars @sekyondaMeta @AlannaBurke

j-silv committed:
Fix #155018 (convert distributed rst to markdown)

Only do git mv for 2 of these files to maintain git log
next commit will be to actually make the conversion changes

j-silv commented:
@pytorchbot label topic: not user facing

j-silv commented:
@pytorchbot label topic: docs

j-silv commented:
@pytorchbot label "topic: docs"

svekars commented on docs/source/distributed.optim.md:
````diff
@@ -1,11 +1,15 @@
+Distributed optimizer is not currently supported when using CUDA tensors
+:::
 
+```{eval-rst}
````
The build [issue](https://github.com/pytorch/pytorch/actions/runs/15576838598/job/43911764169?pr=155528#step:11:828) is in the docstring itself. Need to change this [link](https://github.com/pytorch/pytorch/blob/458cc7213baf01ab742a2250c7a31f7796fcb1ef/torch/distributed/optim/zero_redundancy_optimizer.py#L287) to 
```
 `ZeRO <https://arxiv.org/abs/1910.02054>`_
 ```
 and delete this [line](https://github.com/pytorch/pytorch/blob/458cc7213baf01ab742a2250c7a31f7796fcb1ef/torch/distributed/optim/zero_redundancy_optimizer.py#L369C8-L369C47). 
 Need to check this page in the preview build: 
 https://docs.pytorch.org/docs/main/distributed.optim.html#torch.distributed.optim.ZeroRedundancyOptimizer

j-silv committed:
Fix #155018 - distributed rst to markdown 2nd commit

To maintain git log, this is the second commit for distributed.elastic.md
and distributed.pipelining.md. This is split into 2 to conserve git log history
for the rst files

Also note that we changed zeroredundancy optimizer because of broken target link 'zero'

wz337 reviewed (approved):
LGTM! Thanks!

svekars commented on docs/source/distributed.checkpoint.md:
````diff
@@ -1,110 +1,164 @@
   :members:
+```
 
 In addition to the above entrypoints, `Stateful` objects, as described below, provide additional customization during saving/loading
````
this was broken before, but might as well fix it:

```suggestion
In addition to the above entrypoints, `Stateful` objects, as described below, provide additional customization during saving/loading

```

svekars replied:
Yes, that should work too.

svekars commented:
Now there is a conflict - can you please resolve?

j-silv commented:
> Now there is a conflict - can you please resolve?

Yes, but I won't have access to my computer until tomorrow.

svekars reviewed (approved):
LGTM!

svekars commented:
@pytorchbot merge -f "Docs only change, doc signal is green."
